icu4c/source/i18n/units_router.cpp

```diff
@@ -43,8 +43,24 @@ Precision UnitsRouter::parseSkeletonToPrecision(icu::UnicodeString precisionSkel
     return result;
 }
 
+UnitsRouter::UnitsRouter(const char *inputUnitIdentifer, StringPiece region, StringPiece usage,
+                         UErrorCode &status) {
+    auto inputUnit = MeasureUnit::forIdentifier(inputUnitIdentifer, status);
+    this->init(inputUnit, region, usage, status);
+}
+
 UnitsRouter::UnitsRouter(MeasureUnit inputUnit, StringPiece region, StringPiece usage,
                          UErrorCode &status) {
+    this->init(inputUnit, region, usage, status);
```

Comment:
Can you `std::move(inputUnit)` to make it more explicit that you're not performing a copy?

Reply:
done

Reply:
(Unnecessary to std::move if it's changed to a const reference.)